staker/signing.go

```diff
@@ -0,0 +1,298 @@
+package staker
+
+import (
+	"fmt"
+
+	staking "github.com/babylonlabs-io/babylon/v3/btcstaking"
+	bbntypes "github.com/babylonlabs-io/babylon/v3/types"
+	btcstktypes "github.com/babylonlabs-io/babylon/v3/x/btcstaking/types"
+	cl "github.com/babylonlabs-io/btc-staker/babylonclient"
+	"github.com/babylonlabs-io/btc-staker/walletcontroller"
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+)
+
+type stakeExpSignInfo struct {
+	StakingOutput *wire.TxOut
+	FundingOutput *wire.TxOut
+	SpendInfo     *staking.SpendInfo
+	// Params contains the staking parameters for the previous delegation
+	Params *cl.StakingParams
+}
+
+// signStakingTransaction signs a staking transaction, handling both regular staking
+// and stake expansion transactions with different signing requirements
+func (app *App) signStakingTransaction(tx *wire.MsgTx) (*wire.MsgTx, error) {
```

Comment:
Given that this is called only in `activateVerifiedDelegation` and after we already retrieved the corresponding delegation, maybe we can pass delegation here to deterine whether this is stake expansion or not. This will save us some queries in `signStakeExpansionTransaction`